project-management/src/main/java/life/qbic/projectmanagement/application/api/template/TemplateService.java

```diff
@@ -0,0 +1,292 @@
+   *   <li>Species</li>
+   *   <li>Specimen</li>
+   *   <li>Analyte</li>
+   *   <li>Condition</li>
+   *   <li>Analysis to perform</li>
+   * </ul>
+   *
+   * @param projectId    the project id of the project that contains the experiment for which the
+   *                     template shall be generated
+   * @param experimentId the experiment id of the experiment to create the template for
+   * @return a pre-configured template workbook
+   * @throws NoSuchExperimentException    if no experiment with the provided id can be found.
+   * @throws UnsupportedMimeTypeException if there is no support for the requested {@link MimeType}
+   * @since 1.5.0
+   */
+  @PreAuthorize(
+      "hasPermission(#projectId, 'life.qbic.projectmanagement.domain.model.project.Project', 'READ') ")
+  public DigitalObject sampleUpdateTemplate(String projectId, String experimentId, String batchId,
+      MimeType type) {
+    if (!isSupportedMimeType(type)) {
+      throw new UnsupportedMimeTypeException(UNSUPPORTED_MIME_TYPE + type);
+    }
+    return generateSampleUpdateTemplate(
+        experimentSupplier(projectId, experimentId),
+        projectId,
+        experimentId,
+        batchId);
+  }
+
+  private Supplier<Experiment> experimentSupplier(String projectId, String experimentId) {
+    return () -> experimentService.find(projectId, ExperimentId.parse(experimentId))
+        .orElseThrow(NoSuchExperimentException::new);
+  }
+
+  private DigitalObject generateSampleUpdateTemplate(
+      Supplier<Experiment> experimentSupplier,
+      String projectId,
+      String experimentId,
+      String batchId) {
+    var experiment = experimentSupplier.get();
+    var sampleBasic = querySampleBasicInfo(experiment, projectId, experimentId);
+    var sampleExtension = querySampleExtension(experiment, projectId, experimentId);
+    var samplesInBatch = sampleExtension.samples().stream().filter(inBatchPredicate(batchId))
+        .toList();
+    if (samplesInBatch.isEmpty()) {
+      log.warn("No samples found for experiment during template generation: " + experimentId);
+    }
+
+    return templateProvider.getTemplate(new SampleUpdate(
+        new SampleInformation(
+            samplesInBatch,
+            sampleBasic.analysisMethods(),
+            sampleBasic.conditions(),
+            sampleBasic.analytes(),
+            sampleBasic.species(),
+            sampleBasic.specimen(),
+            sampleExtension.experimentalGroups(),
+            sampleBasic.confoundingVariables(),
+            sampleExtension.confoundingVariableLevels())
+    ));
+  }
+
+  private SampleExtension querySampleExtension(Experiment experiment, String projectId,
+      String experimentId) {
+    var experimentalGroups = experiment.getExperimentalGroups();
+    var samples = sampleService.retrieveSamplesForExperiment(
+        ProjectId.parse(projectId), experimentId).stream().toList();
+    var confoundingVariablesIds = confVariableService.listConfoundingVariablesForExperiment(
```

Comment:
I don't think you need the confounding variables themselves? They are not used in the returned SampleExtension as far as I can see. I think they are part of your basic information and the sample extension only holds the levels of the samples.

Reply:
They are required to make the call to the service to fetch the levels :)

Reply:
Ah you are right